lib/relationships/relationships.ts

```diff
@@ -0,0 +1,279 @@
+import * as _ from 'lodash';
+import { LinkConstraint } from '@balena/jellyfish-client-sdk/build/types';
+import {
+	constraints as sdkLinkConstraints,
+	getReverseConstraint,
+} from '@balena/jellyfish-client-sdk/build/link-constraints';
```

Comment:
We shouldn't bring in all sdk link constraints into the worker. The SDK link constraints include contract types such as `support-thread`, `rating`, etc that the worker doesn't know about and shouldn't need to know about. Each library should only define the relationships related to the contracts that it defines. For example, `plugin-default` defines the `support-thread` and `rating` contract types, so it should be the one that provides the relationships between those two contracts. I think we will need to update the plugin interface to allow plugins to provide an array of relationships in the same way that they can provide an array of contracts and actions. Then the worker will pull those relationships in from all plugins and upsert them (in the same way it does for contracts/actions already).

Reply:
Refactored the relationships to the modules where the contract types are defined